sixtyfps_compiler/llr/lower_expression.rs

```diff
@@ -265,6 +269,25 @@ fn lower_assignment(
 
             llr_Expression::ModelDataAssignment { level, value }
         }
+        tree_Expression::ArrayIndex { array, index } => {
+            let rhs = lower_expression(rhs, ctx);
+            let array = Box::new(lower_expression(array, ctx));
+            let index = Box::new(lower_expression(index, ctx));
+            let value = Box::new(if op == '=' {
+                rhs
+            } else {
+                // FIXME: this will compute the index and the array twice:
+                // Ideally we should store the index and the array in local variable
```

Comment:
Sort of related to this comment: I wonder if we should do the calls to track_row_data_changes() (as well as track_row_count_changes()) in the llr?

AFAICS it would require us to have scoped local variables.

Reply:
I think this is not so frequent that we do `foo[xxx] += something;`
Maybe it is more usual to do `foo[xxx].something += something;` which is a different code path.
We can optimize that later if we need to.

Reply:
True, optimisation wise this may not be worth it. I was thinking more in terms of complexity in the code generators (all three need to know about when exactly to call the two tracking methods). Anyway, can be done later :-)